app/controllers/addapplication/AddApplication.scala

```diff
@@ -78,7 +82,16 @@ class AddApplication @Inject() (
   }
   
   def addApplicationPrincipal(): Action[AnyContent] = loggedInAction { implicit request => 
-    successful(Ok(addApplicationStartPrincipalView())) 
+    
+    val useNewUpliftJourney: Boolean = request.headers.get("useNewUpliftJourney").fold(false) { setting =>
```

Comment:
either make this lazy or change the match below to match on both values so that it works like a truth table

Reply:
changes made